src/schema_factory.ts

```diff
@@ -31,53 +64,28 @@ export function schemaFactory(target: Function) {
         throw new SchemaFactoryError('The argument of schemaFactory should be annotated @Schema() decorator', SchemaFactoryErrorType.NO_SCHEMA_ANNOTATION);
     }
 
-    let queryFieldsDict: any = {};
-    let rootQueryObject: any;
-    if (!Reflect.hasMetadata(GQ_QUERY_KEY, target.prototype)) {
-        throw new SchemaFactoryError('Target should has @Query field', SchemaFactoryErrorType.NO_QUERY_FIELD);
-    } else {
-        const queryMetadata = Reflect.getMetadata(GQ_QUERY_KEY, target.prototype);
-        const queryFields: {[key: string]: any} = {};
-        let queryTypeFn: Function;
-        let queryFieldMetadataList: FieldTypeMetadata[];
-
-        queryMetadata.forEach((queryKey: any) => {
-            queryTypeFn = Reflect.getMetadata('design:type', target.prototype, queryKey) as Function;
-            queryFieldMetadataList = Reflect.getMetadata(GQ_FIELDS_KEY, queryTypeFn.prototype) as FieldTypeMetadata[];
-            queryFieldMetadataList.forEach(def => {
-                queryFields[def.name] = fieldTypeFactory(queryTypeFn, def);
-            });
-        });
+    let rootQueryObject = getRootObject(target, GQ_QUERY_KEY, true, queryObjectTypeFactory);
+    let rootMutationObject = getRootObject(target, GQ_MUTATION_KEY, false, mutationObjectTypeFactory);
+    let rootSubscriptionObject = getRootObject(target, GQ_SUBSCRIPTION_KEY, false, subscriptionObjectTypeFactory, true);
 
-        rootQueryObject = queryObjectTypeFactory(queryFields);
-    }
 
-    let mutationFieldsDict: any = {};
-    if (!Reflect.hasMetadata(GQ_MUTATION_KEY, target.prototype)) {
-        const ret = new graphql.GraphQLSchema({
-            query: rootQueryObject,
-        });
-        return ret;
-    } else {
-        let mutationMetadata = Reflect.getMetadata(GQ_MUTATION_KEY, target.prototype);
-        const fields: {[key: string]: any} = {};
-        let mutationTypeFn: Function;
-        let fieldMetadataList: FieldTypeMetadata[];
+    // let result: any = {
```

Comment:
Could you remove this code that is commented out?

Reply:
done